nmigen_boards/ecp5_5g_evn.py

```diff
@@ -15,20 +15,117 @@ class ECP55GEVNPlatform(LatticeECP5Platform):
+                         attrs=Attrs(IO_TYPE=VCCIO6)),
+        
+        *SwitchResources(pins={4: "E15", 5: "D16", 6: "B16", 7: "C16", 8: "A16"}, invert=True,
+                         attrs=Attrs(IO_TYPE=VCCIO1)),
+        
+        # SERDES
+        Resource("serdes", 0,
+            Subsignal("tx", DiffPairs("W4", "W5", dir="o")),
+            Subsignal("rx", DiffPairs("Y5", "Y6", dir="i")),
+        ),
+        Resource("serdes", 1,
+            Subsignal("tx", DiffPairs("W8", "W9", dir="o")),
+            Subsignal("rx", DiffPairs("Y7", "Y8", dir="i")),
+        ),
+        Resource("serdes", 2,
+            Subsignal("tx", DiffPairs("W13", "W14", dir="o")),
+            Subsignal("rx", DiffPairs("Y14", "Y15", dir="i")),
+        ),
+        Resource("serdes", 3,
+            Subsignal("tx", DiffPairs("W17", "W18", dir="o")),
+            Subsignal("rx", DiffPairs("Y16", "Y17", dir="i")),
+        ),
+        
+        Resource("serdesclk", 0, DiffPairs("Y11", "Y12", dir="o")),
+        Resource("serdesclk", 1, DiffPairs("Y19", "W20", dir="o")), # 200 MHz
+        
+        # Diff pairs, they have selectable termination resistors
+        # Resource("diff", "0", DiffPairs(A4, A5), Attrs(IO_TYPE=VCCIO7)),
+        # Resource("diff", "1", DiffPairs(B5, C5), Attrs(IO_TYPE=VCCIO7)),
+        # Resource("diff", "2", DiffPairs(C4, B4), Attrs(IO_TYPE=VCCIO7)),
+        # Resource("diff", "3", DiffPairs(A3, B3), Attrs(IO_TYPE=VCCIO7)),
+        # Resource("diff", "4", DiffPairs(E4, D5), Attrs(IO_TYPE=VCCIO7)),
+        # Resource("diff", "5", DiffPairs(C3, D3), Attrs(IO_TYPE=VCCIO7)),
+        # Resource("diff", "6", DiffPairs(F4, E3), Attrs(IO_TYPE=VCCIO7)),
+        # Resource("diff", "7", DiffPairs(E5, F5), Attrs(IO_TYPE=VCCIO7)),
+        # Resource("diff", "8", DiffPairs(A2, B1), Attrs(IO_TYPE=VCCIO7)),
+        # Resource("diff", "9", DiffPairs(B2, C2), Attrs(IO_TYPE=VCCIO7)),
+        # Resource("diff", "10", DiffPairs(C1, D1), Attrs(IO_TYPE=VCCIO7)),
+        # Resource("diff", "11", DiffPairs(D2, E1), Attrs(IO_TYPE=VCCIO7)),
+        # Resource("diff", "12", DiffPairs(H4, G5), Attrs(IO_TYPE=VCCIO7)),
+        # Resource("diff", "13", DiffPairs(H5, H3), Attrs(IO_TYPE=VCCIO7)),
+        # Resource("diff", "14", DiffPairs(G3, F3), Attrs(IO_TYPE=VCCIO7)),
+        # Resource("diff", "15", DiffPairs(F2, E2), Attrs(IO_TYPE=VCCIO7)),
+        # Resource("diff", "16", DiffPairs(G19, H20, Attrs(IO_TYPE=VCCIO2)),
+        # Resource("diff", "17", DiffPairs(J18, K18, Attrs(IO_TYPE=VCCIO2)),
+        # Resource("diff", "18", DiffPairs(J19, K19, Attrs(IO_TYPE=VCCIO2)),
+        # Resource("diff", "19", DiffPairs(J20, K20, Attrs(IO_TYPE=VCCIO2)),
+        
         # TODO: add other resources
     ]
     connectors  = [
-        # TODO: add connectors
+        Connector("J", 39, "- - - D15 B15 C15 B13 B20 D11 E11 B12 C12 D12 E12 C13 D13 E13 A14 A9 B10 - - - - - - - - E7 - A11 - A19 - - - - - - -"),
```

Comment:
In the previous PR I suggested you leave out the connector changes so that the extra work of cross-checking these during review (which could take a while) would not block merging the rest of the PR. But it's fine to leave it this way too.